spinalcordtoolbox/scripts/sct_run_batch.py

```diff
@@ -175,6 +180,35 @@ def _find_nonsys32_bash_exe():
     return shutil.which('bash', path=nonsys32_paths)
 
 
+def _parse_dataset_directory(path_data, subject_prefix="sub-", ignore_ses=False):
+    """
+    Parse a dataset directory to find subject directories (or session subdirectories, if present).
+
+    Notes:
+        - The dataset is assumed to be structured in a (somewhat) BIDS-compliant way, see:
+          https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#file-name-structure
+        - This function is a rudimentary version of the library PyBIDS: https://github.com/bids-standard/pybids.
+          TODO: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3415
+    """
+    # Find subject directories within the root dataset folder
+    subject_dirs = sorted([f for f in os.listdir(path_data)
+                           if f.startswith(subject_prefix) and os.path.isdir(os.path.join(path_data, f))])
+
+    # Find session subdirectories within subject directories
+    session_dirs = []
+    for sub_dir in subject_dirs:
+        path_sub = os.path.join(path_data, sub_dir)
+        subject_session_dirs = sorted([os.path.join(sub_dir, f) for f in os.listdir(path_sub)
+                                       if f.startswith("ses-") and os.path.isdir(os.path.join(path_sub, f))])
```

Comment:
Thanks for adding the `isdir` check for `ses-` subfolders, which mirrors the check for `sub-` folders.